pkg/logutil/log.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logutil
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/coreos/pkg/capnslog"
+	"github.com/juju/errors"
+	lumberjack "gopkg.in/natefinch/lumberjack.v2"
+)
+
+const (
+	defaultLogTimeFormat = "2006/01/02 15:04:05"
+	defaultLogMaxSize    = 300 // MB
+	defaultLogMaxBackups = 3
+	defaultLogMaxAge     = 28 // days
+	defaultLogLevel      = log.InfoLevel
+
+	logDirMode = 0755
+)
+
+// LogConfig serializes log related config in toml/json.
+type LogConfig struct {
+	// Log level.
+	Level string `toml:"level" json:"level"`
+	// Log file.
+	Filename string `toml:"file" json:"file"`
+}
+
+// redirectFormatter will redirect etcd logs to logrus logs.
+type redirectFormatter struct{}
+
+// Format implements capnslog.Formatter hook.
+func (rf *redirectFormatter) Format(pkg string, level capnslog.LogLevel, depth int, entries ...interface{}) {
+	if pkg != "" {
+		pkg = fmt.Sprint(pkg, ": ")
+	}
+
+	logStr := fmt.Sprint(pkg, entries)
+
+	// trim square brackets
+	if len(logStr) > 2 && logStr[0] == '[' {
+		logStr = logStr[1 : len(logStr)-1]
+	}
```

Comment:
When will it starts with '['? And the last char will always be ']' if first is '['?